### DISCUSSION TITLE
How to add the "allow-cors" flag while starting Appium server in the java program.

### DISCUSSION THREAD

**Discussion by qatestingravi** [appium/appium#20248] [Q&A]
We can use "--allow-cors" flag from the command line during Appium server staring, like "appium --allow-cors", But how to add via java program. Say for example in the below java program,how to add "allow-cors" flag? 

      AppiumDriverLocalService service = buildService(new AppiumServiceBuilder()
	      .withIPAddress("127.0.0.1")
	      .usingPort(4723)
	      .withTimeout(Duration.ofSeconds(120));
	
	  service.start();

**Comment by portyu9** [comment]
Hi @qatestingravi 

You can pass `--allow-cors` through `AppiumServiceBuilder` using a custom `ServerArgument`:

```java
import io.appium.java_client.service.local.flags.ServerArgument;

ServerArgument allowCors = () -> "--allow-cors";

AppiumDriverLocalService service = AppiumDriverLocalService.buildService(
    new AppiumServiceBuilder()
        .withIPAddress("127.0.0.1")
        .usingPort(4723)
        .withArgument(allowCors)
        .withTimeout(Duration.ofSeconds(120))
);

service.start();
```

`withArgument()` is the correct method for boolean Appium server flags—the presence of the argument enables it.

Only enable `--allow-cors` when required, and preferably keep Appium bound to localhost or another trusted interface.

Please mark this answer as accepted if it helped, thank you.
